internal/services/paloalto/schema/source.go

```diff
@@ -0,0 +1,120 @@
+package schema
+
+import (
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/paloaltonetworks/2022-08-29/localrules"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/paloaltonetworks/2022-08-29/prefixlistlocalrulestack"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/paloalto/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type Source struct {
+	CIDRS       []string `tfschema:"cidrs"`
+	Countries   []string `tfschema:"countries"`
+	Feeds       []string `tfschema:"feeds"`
+	PrefixLists []string `tfschema:"prefix_list_ids"`
+}
+
+func SourceSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Required: true,
+		MaxItems: 1,
+		MinItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"cidrs": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+						ValidateFunc: validation.Any(
+							validation.IsCIDR,
+							validation.StringInSlice([]string{"any"}, false),
+						),
+					},
+					AtLeastOneOf: []string{
+						"source.0.cidrs",
+						"source.0.countries",
+						"source.0.feeds",
+						"source.0.prefix_list_ids",
+					},
+				},
+
+				"countries": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Schema{
+						Type:         pluginsdk.TypeString,
+						ValidateFunc: validate.ISO3361CountryCode,
+					},
+					AtLeastOneOf: []string{
+						"source.0.cidrs",
+						"source.0.countries",
+						"source.0.feeds",
+						"source.0.prefix_list_ids",
+					},
+				},
+
+				"feeds": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Schema{
+						Type:         pluginsdk.TypeString,
+						ValidateFunc: nil, // TODO - This is another resource type?
+					},
+					AtLeastOneOf: []string{
+						"source.0.cidrs",
+						"source.0.countries",
+						"source.0.feeds",
+						"source.0.prefix_list_ids",
+					},
+				},
+
+				"prefix_list_ids": {
```

Comment:
tbh we should probably prefix these ID types with
```suggestion
				"local_rulestack_prefix_list_ids": {
```